builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -128,14 +129,18 @@ func resourceCloudStackInstanceCreate(d *schema.ResourceData, meta interface{})
 	p := cs.VirtualMachine.NewDeployVirtualMachineParams(serviceofferingid, templateid, zone.Id)
 
 	// Set the name
-	name := d.Get("name").(string)
-	p.SetName(name)
+	name, hasName := d.GetOk("name")
+	if hasName {
+		p.SetName(name.(string))
+	}
 
 	// Set the display name
 	if displayname, ok := d.GetOk("display_name"); ok {
 		p.SetDisplayname(displayname.(string))
 	} else {
-		p.SetDisplayname(name)
+		if hasName {
```

Comment:
Think this would be just a little cleaner:

```go
// Set the display name
if displayname, ok := d.GetOk("display_name"); ok {
  p.SetDisplayname(displayname.(string))
} else if hasName {
  p.SetDisplayname(name.(string))
}
```